### ISSUE TITLE
hugging face demo does not work in my local environment

### ISSUE THREAD

**Issue by yolk-pie-L** [kserve/kserve#3516]
I followed the guide here https://github.com/kserve/kserve/tree/master/python/huggingfaceserver. I tried run hugging face server locally, but it doesn't work.

**My running process is as follows:**

First, create a conda environment:
```
conda create -n kserve python=3.10
```

Then, install the necessary packages:
```
C:\Users\myusername\AppData\Roaming\Python\Scripts\poetry install
```

After that, run this command
```
 python -m huggingfaceserver --model_id=bert-base-uncased --model_name=bert
```

**Error message**

```
(kserve) PS D:\GithubRepo\kserve\python\huggingfaceserver> python -m huggingfaceserver --model_id=bert-base-uncased --model_name=bert
D:\AppData\Miniconda3\envs\kserve\lib\site-packages\timing_asgi\utils.py:9: UserWarning: resource module not available for Windows, cpu time will be set to 0
  warnings.warn("resource module not available for Windows, cpu time will be set to 0")
INFO:kserve:successfully loaded tokenizer for task: 5
Some weights of the model checkpoint at bert-base-uncased were not used when initializing BertForMaskedLM: ['bert.pooler.dense.bias', 'bert.pooler.dense.weight', 'cls.seq_relationship.bias', 'cls.seq_relationship.weight']
- This IS expected if you are initializing BertForMaskedLM from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertForMaskedLM from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
INFO:kserve:successfully loaded huggingface model from path bert-base-uncased
INFO:kserve:Registering model: bert
INFO:kserve:Setting max asyncio worker threads as 24
INFO:kserve:Starting uvicorn with 1 workers
INFO:kserve:Starting gRPC server on [::]:8081
Traceback (most recent call last):
  File "D:\AppData\Miniconda3\envs\kserve\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "D:\AppData\Miniconda3\envs\kserve\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "D:\GithubRepo\kserve\python\huggingfaceserver\huggingfaceserver\__main__.py", line 70, in <module>
    kserve.ModelServer().start([model] if model.ready else [])
  File "D:\GithubRepo\kserve\python\kserve\kserve\model_server.py", line 226, in start
    asyncio.run(servers_task())
  File "D:\AppData\Miniconda3\envs\kserve\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "D:\AppData\Miniconda3\envs\kserve\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "D:\GithubRepo\kserve\python\kserve\kserve\model_server.py", line 224, in servers_task
    await asyncio.gather(*servers)
  File "D:\GithubRepo\kserve\python\kserve\kserve\model_server.py", line 188, in serve
    loop.add_signal_handler(
  File "D:\AppData\Miniconda3\envs\kserve\lib\asyncio\events.py", line 553, in add_signal_handler
    raise NotImplementedError
NotImplementedError
Traceback (most recent call last):
  File "D:\AppData\Miniconda3\envs\kserve\lib\site-packages\grpc\aio\_server.py", line 185, in __del__
  File "src\python\grpcio\grpc\_cython\_cygrpc/aio/common.pyx.pxi", line 120, in grpc._cython.cygrpc.schedule_coro_threadsafe   
  File "src\python\grpcio\grpc\_cython\_cygrpc/aio/common.pyx.pxi", line 112, in grpc._cython.cygrpc.schedule_coro_threadsafe   
  File "D:\AppData\Miniconda3\envs\kserve\lib\asyncio\base_events.py", line 436, in create_task
  File "D:\AppData\Miniconda3\envs\kserve\lib\asyncio\base_events.py", line 515, in _check_closed
RuntimeError: Event loop is closed
sys:1: RuntimeWarning: coroutine 'AioServer.shutdown' was never awaited
```

**Environment:**

My OS is Windows. My poetry version is `Poetry (version 1.8.2)`.

**Comment by yolk-pie-L** [comment]
I think it is because it only supports linux-based OS. Hope the documents can clarify it.

**Comment by AlexanderSing** [comment]
From what I understand, this should've been fixed with #3089 but from what I can tell from the Python documentation, asyncio loops don't support the "add_signal_handler()" method at all (see https://docs.python.org/3/library/asyncio-platforms.html)
The code suggested in #2915 would've worked as it catches the non-availability of the function but the fix implemented in #3089 does not. As a workaround one can either use the code from #2915 or simply set sig_list to an empty list on windows.

From #2915 :
```
# SIGQUIT is not available on Windows
try:
    sig_list = [signal.SIGINT, signal.SIGTERM, signal.SIGQUIT]
except:
    sig_list = [signal.SIGINT, signal.SIGTERM]

# add_signal_handler may not have been implemented
try:
    for sig in sig_list:
        loop.add_signal_handler(
            sig, lambda s=sig: asyncio.create_task(self.stop(sig=s))
        )
except:
    pass
```

or :

```
if sys.platform not in ['win32', 'win64']:
                sig_list = [signal.SIGINT, signal.SIGTERM, signal.SIGQUIT]
            else:
                sig_list = []
```
